doc/utils/manifest.md

```diff
@@ -47,6 +50,17 @@ This section is optional. It contains metadata used to run application tests.
 | `unit_directory`   | Path of the directory where unit tests can be found                                            |
 | `pytest_directory` | Path of the directory where functional, Python test can be found (`conftest.py` file expected) |
 
+#### `[build_options]`
+
+This section is optional. It contains metadata helping select build options depending on use cases
+
+| Field name   | Description |
+|--------------|-------------|
+| `<use_case>` | `OPTIONS`   |
```

Comment:
Maybe explain a bit more what should be "OPTIONS" (and where it will be used). I guess it will also depend on whether it is a rust or C app. 

For a C app it will be (I guess) env variables passed before calling make : `<OPTIONS> && make` but for Rust apps it will be passed as a cargo option `cargo ledger build nanos -- <OPTIONS>`